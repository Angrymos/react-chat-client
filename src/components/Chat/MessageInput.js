import React from 'react';
import classNames from 'classnames';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import Paper from '@material-ui/core/Paper';

const styles = theme => ({
  messageInput: {
    padding: theme.spacing.unit * 2,
  },
});

const MessageInput = ({ classes }) => (
  <Paper className={classes.messageInput} elevation={6}>
    <Input
      fullWidth
      placeholder='Type your messageâ€¦'
    />
  </Paper>
);

export default withStyles(styles)(MessageInput);
