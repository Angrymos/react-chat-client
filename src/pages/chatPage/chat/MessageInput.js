import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Input from '@material-ui/core/Input';
import Paper from '@material-ui/core/Paper';
import Button from '../../../../node_modules/@material-ui/core/Button/Button';

const styles = theme => ({
  messageInputWrapper: {
    position: 'fixed',
    left: 'auto',
    right: 32,
    bottom: 0,
    width: 'calc(100% - 352px)',
    padding: theme.spacing.unit * 3,
  },

  messageInput: {
    padding: theme.spacing.unit * 2,
  },
});

class MessageInput extends React.Component {
  state = {
    value: '',
  };

  handleOnChangeInput = (event) => {
    this.setState({
      value: event.target.value,
    });
  };

  handleOnKeyPress = (event) => {
    const { value } = this.state;

    if (event.key === 'Enter' && value) {
      this.props.sendMessage(value);
      this.setState({ value: '' });
    }
  };

  render() {
    const { classes, onClickJoin, showJoinButton } = this.props;
    return (
      <div className={classes.messageInputWrapper}>
        <Paper className={classes.messageInput} elevation={6}>
          {showJoinButton ? (
            <Button
              fullWidth
              variant='raised'
              color='primary'
              onClick={onClickJoin}
            >
              Join
            </Button>
          ) : (
            <Input
              fullWidth
              placeholder='Type your messageâ€¦'
              value={this.state.value}
              onChange={this.handleOnChangeInput}
              onKeyPress={this.handleOnKeyPress}
            />
          )}
        </Paper>
      </div>
    );
  }
}

export default withStyles(styles)(MessageInput);
