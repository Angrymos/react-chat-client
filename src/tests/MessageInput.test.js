import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageInput from '../pages/chatPage/chat/MessageInput';

const mockProps = {
  disabled: false,
  showJoinButton: false,

  onClickJoin: jest.fn(),
  sendMessage: jest.fn(),
};

describe('<MessageInput />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageInput {...mockProps} />, div);
  });

  it('renders input', () => {
    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled input', () => {
    const props = {
      ...mockProps,
      disabled: true,
    };

    const tree = renderer.create(<MessageInput {...props}  />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders button', () => {
    const props = {
      ...mockProps,
      showJoinButton: true,
    };

    const tree = renderer.create(<MessageInput {...mockProps}  />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders disabled button', () => {
    const props = {
      ...mockProps,
      disabled: true,
      showJoinButton: true,
    };

    const tree = renderer.create(<MessageInput {...mockProps} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
