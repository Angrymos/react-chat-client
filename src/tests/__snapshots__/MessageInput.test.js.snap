// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageInput /> renders button 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> renders disabled button 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> renders disabled input 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-disabled-46 MuiInput-disabled-33 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-disabled-46 MuiInput-disabled-33"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput /> renders input 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput {...mockProps}/> renders button 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput {...mockProps}/> renders disabled button 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput {...mockProps}/> renders disabled input 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-disabled-46 MuiInput-disabled-33 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38 MuiInputBase-disabled-46 MuiInput-disabled-33"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`<MessageInput {...mockProps}/> renders input 1`] = `
<div
  className="MessageInput-messageInputWrapper-1"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation6-11 MuiPaper-rounded-4 MessageInput-messageInput-2"
  >
    <div
      className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-fullWidth-52 MuiInput-fullWidth-37"
      onClick={[Function]}
      onKeyPress={[Function]}
    >
      <input
        className="MuiInputBase-input-53 MuiInput-input-38"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Type your message…"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
